libyul/optimiser/Suite.cpp

```diff
@@ -429,6 +246,60 @@ map<char, string> const& OptimiserSuite::stepAbbreviationToNameMap()
 	return lookupTable;
 }
 
+void OptimiserSuite::runSequence(string const& _stepAbbreviations, Block& _ast)
+{
+	string input = _stepAbbreviations;
+	boost::remove_erase(input, ' ');
+	boost::remove_erase(input, '\n');
+
+	bool insideLoop = false;
+	for (char abbreviation: input)
+		switch (abbreviation)
+		{
+		case '(':
+			assertThrow(!insideLoop, OptimizerException, "Nested parentheses not supported");
+			insideLoop = true;
+			break;
+		case ')':
+			assertThrow(insideLoop, OptimizerException, "Unbalanced parenthesis");
+			insideLoop = false;
+			break;
+		default:
+			assertThrow(
+				stepAbbreviationToNameMap().find(abbreviation) != stepAbbreviationToNameMap().end(),
+				OptimizerException,
+				"Invalid optimisation step abbreviation"
+			);
+		}
+	assertThrow(!insideLoop, OptimizerException, "Unbalanced parenthesis");
+
+	auto abbreviationsToSteps = [](string const& _fullSequence, size_t _begin, size_t _end) -> vector<string>
+	{
+		vector<string> steps;
+		for (char abbreviation: _fullSequence.substr(_begin, _end - _begin))
+			steps.push_back(stepAbbreviationToNameMap().at(abbreviation));
+
+		return steps;
+	};
```

Comment:
If you run `substr` anyway, maybe just pull that out, i.e. change the function to
```suggestion
	auto abbreviationsToSteps = [](string const& _sequence) -> vector<string>
	{
		vector<string> steps;
		for (char abbreviation: _sequence)
			steps.emplace_back(stepAbbreviationToNameMap().at(abbreviation));
		return steps;
	};
```

Reply:
Done.

I put `substr()` in the function because I wanted to move as much of the repeating code out of the loop as possible. But you're right. This particular bit did not save us all that much apart from making the `runSequence()` calls a little shorter.